spec/models/spree/product_spec.rb

```diff
@@ -321,13 +257,16 @@ module Spree
       let(:product) { create(:simple_product) }
 
       describe "touching affected enterprises when the product is deleted" do
-        let(:product) { create(:simple_product, supplier: distributor) }
-        let(:supplier) { product.supplier }
+        let(:product) { create(:simple_product) }
+        let(:supplier) { create(:supplier_enterprise) }
         let(:distributor) { create(:distributor_enterprise) }
         let!(:oc) {
           create(:simple_order_cycle, distributors: [distributor],
                                       variants: [product.variants.first])
         }
+        before do
+          create(:variant, product:, supplier:)
+        end
 
         it "touches the supplier" do
```

Comment:
I think this test isn't needed on product now, and could be moved to variant, removing the need for the above adjustments.

Reply:
Indeed, the variant being deleted when a product is deleted we can then touch the supplier when the variant is deleted, I'll fix that.